File: src/main/kotlin/introductory_problems/Repetitions.java
package introductory_problems;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Repetitions {
    static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String dnaStrand = br.readLine().trim();

        int answer = solution(dnaStrand);
        System.out.println(answer);
    }

    private static int solution(String dnaStrand) {
        char previous = 0;
        int current = 0;
        int max = 0;

        for (char block : dnaStrand.toCharArray()) {
            if (previous != 0 && previous == block) {
                current++;
            } else {
                current = 1;
            }

            if (current > max) {
                max = current;
            }

            previous = block;
        }

        return max;
    }
}

File: src/main/kotlin/cses/introductory_problems/Repetitions.kt
package cses.introductory_problems

fun main() {
//    val dnaStrand = readln()
    val dnaStrand = "ATTCGGGA"

    val answer = solution(dnaStrand)

    println(answer)
}

private fun solution(dnaStrand: String): Int {
    var previous: Char? = null
    var current = 0
    var max = 0

    dnaStrand.forEach { block ->
        if (previous != null && previous == block) current++
        else current = 1
        if (current > max) max = current
        previous = block
    }

    return max
}
